web/modules/custom/dpl_paragraphs/dpl_paragraphs.module

```diff
@@ -108,3 +109,56 @@ function dpl_paragraphs_preprocess_paragraph__medias(array &$variables): void {
   $alternative_images = $paragraph->get('field_medias')->view('alternative');
   $variables['content']['field_medias'][1] = $alternative_images[1] ?? NULL;
 }
+
+/**
+ * Implements hook_preprocess_paragraph__TYPE().
+ *
+ * Prepares variables for paragraph--recommendation.html.twig template.
+ *
+ * If the paragraph has a title or description, we use the recommended-material
+ * React app. Otherwise, we use the recommendation React app.
+ */
+function dpl_paragraphs_preprocess_paragraph__recommendation(array &$variables): void {
+  /** @var \Drupal\paragraphs\ParagraphInterface $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  if (!$paragraph->hasField('field_work_id') || $paragraph->get('field_work_id')->isEmpty()) {
+    return;
+  }
+  $work_id = $paragraph->get('field_work_id')->value;
+  $paragraph_title = $paragraph->get('field_recommendation_title')->value;
+  $paragraph_description = $paragraph->get('field_recommendation_description')->value;
+  $paragraph_image_position_right = $paragraph->get('field_image_position_right')->value;
+
+  // Determine which React app to use.
+  if (!empty($paragraph_title) || !empty($paragraph_description)) {
+
+    // The URL is only needed for the recommended-material app.
+    $base_url = \Drupal::request()->getSchemeAndHttpHost();
```

Comment:
This should not be necessary.
You're looking for what Drupal calls an "absolute URL":

https://simpledrupal.com/article/how-create-absolute-url-route-name

Reply:
Yeah this is defo better 🪂